drivers/display/display_ili9340.c

```diff
@@ -92,50 +126,124 @@ int ili9340_init(struct device *dev)
 	LOG_DBG("Exiting sleep mode");
 	ili9340_exit_sleep(data);
 
-	/* device_get_binding checks if driver_api is not zero before checking
-	 * device name.
-	 * So just set driver_api to 1 else the function call will fail
-	 */
-	dev->driver_api = (void *)1;
-
 	return 0;
 }
 
-void ili9340_write_pixel(const struct device *dev, const u16_t x, const u16_t y,
-			 const u8_t r, const u8_t g, const u8_t b)
+int ili9340_write(const struct device *dev, const u16_t x,
+		  const u16_t y, const u16_t w, const u16_t h,
+		  const u16_t p, const void *pix_data)
 {
-	u8_t rgb_data[] = {r, g, b};
+	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+	const void *write_data_start = pix_data;
+	struct spi_buf tx_buf;
+	struct spi_buf_set tx_bufs;
+	u16_t write_cnt;
+	u16_t nbr_of_writes;
+	u16_t write_h;
+
+	LOG_DBG("Writing %dx%d (w,h) @ %dx%d (x,y)", w, h, x, y);
+	ili9340_set_mem_area(data, x, y, w, h);
+
+	__ASSERT(w >= p, "Pitch is smaller then width");
+	if (p > w) {
+		write_h = 1;
+		nbr_of_writes = h;
+	} else {
+		write_h = h;
+		nbr_of_writes = 1;
+	}
+
+	ili9340_transmit(data, ILI9340_CMD_MEM_WRITE,
+				(void *) write_data_start, 3 * w * write_h);
+
+	tx_bufs.buffers = &tx_buf;
+	tx_bufs.count = 1;
+
+	for (write_cnt = 1, write_data_start += (3 * p);
```

Comment:
It will be easier to read if you do not hide `write_data_start += (3 * p)` here.